Fix googletest lookup and add missing (pkg-config) linker flags

--- ament_cmake_gtest/ament_cmake_gtest-extras.cmake~	2022-02-18 01:03:31.000000000 +0100
+++ ament_cmake_gtest/ament_cmake_gtest-extras.cmake	2022-03-30 22:49:41.228850204 +0200
@@ -103,6 +103,11 @@
       set(GTEST_LIBRARIES ${GTEST_FROM_SOURCE_LIBRARIES})
       set(GTEST_MAIN_LIBRARIES ${GTEST_FROM_SOURCE_MAIN_LIBRARIES})
       set(GTEST_BOTH_LIBRARIES ${GTEST_LIBRARIES} ${GTEST_MAIN_LIBRARIES})
+    else()
+      find_package(PkgConfig REQUIRED)
+      pkg_check_modules(GTEST REQUIRED gtest)
+      pkg_check_modules(GTEST_MAIN REQUIRED gtest_main)
+      set(GTEST_BOTH_LIBRARIES ${GTEST_LIBRARIES} ${GTEST_MAIN_LIBRARIES})
     endif()
 
     if(NOT GTEST_FOUND)
--- ament_cmake_gtest/cmake/ament_add_gtest_executable.cmake~	2022-02-18 01:03:31.000000000 +0100
+++ ament_cmake_gtest/cmake/ament_add_gtest_executable.cmake	2022-03-30 23:01:30.780740150 +0200
@@ -50,8 +50,10 @@
   add_executable("${target}" ${ARG_UNPARSED_ARGUMENTS})
   target_include_directories("${target}" BEFORE PUBLIC "${GTEST_INCLUDE_DIRS}")
   if(NOT ARG_SKIP_LINKING_MAIN_LIBRARIES)
+    target_link_libraries("${target}" ${GTEST_MAIN_LDFLAGS})
     target_link_libraries("${target}" ${GTEST_MAIN_LIBRARIES})
   endif()
+  target_link_libraries("${target}" ${GTEST_LDFLAGS})
   target_link_libraries("${target}" ${GTEST_LIBRARIES})
   if(NOT WIN32)
     set(THREADS_PREFER_PTHREAD_FLAG ON)
